python/orca/src/bigdl/orca/learn/tf2/spark_runner.py

```diff
@@ -215,8 +211,10 @@ def __init__(self, model_creator, compile_args_creator,
         self.setup()
         self.cluster = cluster_info
         if mode == "fit" or mode == "evaluate":
+            from pyspark import BarrierTaskContext
             self.partition_id = BarrierTaskContext.get().partitionId()
         else:
+            from pyspark import TaskContext
```

Comment:
Why can't we always use TaskContext to get partitionId?

Reply:
Seems we can?

Reply:
Tested, we can't do this.
When call rdd.barrier(), TaskContext.get() will return None, only BarrierTaskContext will return the actual instance.